src/lib/prover/prover.ml

```diff
@@ -251,9 +275,22 @@ let create ~logger ~pids =
       [ ("prover_pid", `Int (Process.pid process |> Pid.to_int))
       ; ( "process_kind"
         , `String Child_processes.Termination.(show_process_kind Prover) ) ] ;
-  Child_processes.Termination.(register_process pids process Prover) ;
-  File_system.dup_stdout process ;
-  File_system.dup_stderr process ;
+  Child_processes.Termination.register_process pids process
+    Child_processes.Termination.Prover ;
+  don't_wait_for
+  @@ Pipe.iter
+       (Process.stdout process |> Reader.pipe)
+       ~f:(fun stdout ->
+         return
+         @@ Logger.debug logger ~module_:__MODULE__ ~location:__LOC__
```

Comment:
Won't this double-wrap the child structured JSON logs in a new structured JSON message? This is a bug that we've had to fix before to have the logging infrastructure be happy.